src/parsers/babel_parser.js

```diff
@@ -30,6 +30,20 @@ export const parse = (file: string, data: ?string, options: ?parser.ParserOption
   const parseResult = new ParseResult(file);
   const [ast, _data] = _getASTfor(file, data, options);
 
+  const getRootOfTypeOrSelf = (node, type) => {
+    let rootForType = node;
+    while (rootForType[type]) {
+      rootForType = rootForType[type];
+    }
+    return rootForType;
+  };
+
+  const getRootCalleeOrSelf = node => getRootOfTypeOrSelf(node, 'callee');
+
+  const getRootTagOrSelf = node => getRootOfTypeOrSelf(node, 'tag');
+
+  const getRootObjectOrSelf = node => getRootOfTypeOrSelf(node, 'object');
```

Comment:
I like your approach but it seems that we could make it more expressive...

```suggestion
  const deepGet = (node, ...types: string[]) => 
    types.reduce<BabelNode>((rootForType, type) => {
      while (rootForType[type]) {
        rootForType = rootForType[type];
      }
      return rootForType;
    }, node);
```

so we can call this with clear order below...

Reply:
> I like your approach but it seems that we could make it more expressive...
> 
> so we can call this with clear order below...

I like this suggestion.